File: tests/human/x/ex/group_by_join.ex
customers = [
  %{id: 1, name: "Alice"},
  %{id: 2, name: "Bob"}
]

orders = [
  %{id: 100, customerId: 1},
  %{id: 101, customerId: 1},
  %{id: 102, customerId: 2}
]

stats =
  orders
  |> Enum.map(fn o ->
    customer = Enum.find(customers, &(&1.id == o.customerId))
    Map.put(o, :customer, customer)
  end)
  |> Enum.group_by(& &1.customer.name)
  |> Enum.map(fn {name, group} -> %{name: name, count: Enum.count(group)} end)

IO.puts("--- Orders per customer ---")
Enum.each(stats, fn s ->
  IO.puts("#{s.name} orders: #{s.count}")
end)

File: tests/transpiler/x/fs/group_by_join.fs
// Generated 2025-07-21 15:37 +0700
open System

type Anon1 = {
    mutable id: int
    mutable name: string
}
type Anon2 = {
    mutable id: int
    mutable name: string
}
type Anon3 = {
    mutable id: int
    mutable customerId: int
}
type Anon4 = {
    mutable id: int
    mutable customerId: int
}
type Anon5 = {
    mutable name: obj
    mutable count: int
}
type Anon6 = {
    mutable name: obj
    mutable count: int
}
let customers: Anon2 list = [{ id = 1; name = "Alice" }; { id = 2; name = "Bob" }]
let orders: Anon4 list = [{ id = 100; customerId = 1 }; { id = 101; customerId = 1 }; { id = 102; customerId = 2 }]
let stats: Anon6 list = [ for (key, items) in List.groupBy (fun { o = o; c = c } -> c.name) [ for o in orders do for c in customers do if (o.customerId) = (c.id) then yield {| o = o; c = c |} ] do
    let g = {| key = key; items = items |}
    yield { name = g.key; count = List.length (g.items) } ]
printfn "%s" (string "--- Orders per customer ---")
for s in stats do
printfn "%s" (String.concat " " [string (s.name); string "orders:"; string (s.count)])
